File: backend/src/main/java/br/gov/ac/tce/licon/entities/Tecnico.java
package br.gov.ac.tce.licon.entities;

import lombok.*;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.validation.constraints.NotNull;
import java.time.LocalDate;


@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Tecnico extends AbstractAuditableEntity {

	@NotNull
	@Column(name = "NOME")
	private String nome;

	@NotNull
	@Column(name = "NUM_TITULOS")
	private Integer numTitulos;

	@NotNull
	@Column(name = "DATA_NASCIMENTO")
	private LocalDate dataNascimento;

	@NotNull
	@Column(name = "ESTILO_JOGO")
	private String estiloJogo;

}

File: frontend/src/domains/Tecnico.js
import { observable } from 'mobx';
import DomainBase from './DomainBase';

class Tecnico extends DomainBase {
  @observable id;
  @observable nome;
  @observable numTitulos;
  @observable dataNascimento;
  @observable estiloJogo;

  static getDomainAttributes() {
    return ['id', 'nome', 'numTitulos', 'dataNascimento', 'estiloJogo'];
  }
}

export default Tecnico;
